crates/pevm/src/vm.rs

```diff
@@ -79,26 +79,34 @@ pub(crate) enum VmExecutionError {
 }
 
 /// Errors when reading a memory location.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, thiserror::Error)]
 pub enum ReadError {
     /// Cannot read memory location from storage.
+    // TODO: More concreate type
+    #[error("Failed reading memory from storage: {0}")]
     StorageError(String),
     /// This memory location has been written by a lower transaction.
+    #[error("Read of memory location is blocked by tx #{0}")]
     Blocking(TxIdx),
     /// There has been an inconsistent read like reading the same
     /// location from storage in the first call but from [`VmMemory`] in
     /// the next.
+    #[error("Transactions read inconsistent storage slots")]
```

Comment:
The message here is misleading: "read inconsistent storage slots" sounds like "previously read from slot A but now read from slot B".

In fact, the `InconsistentRead` means the key (`Address` or `(Address, B256)`) is the same but the values are from different versions (e.g. one from storage and one from `VmMemory`).

Reply:
Before there is a better message, I think we can write `#[error("Inconsistent read")]`.